src/refactor/connectionManager.py

```diff
@@ -0,0 +1,51 @@
+'''
+ * Copyright (C) 2022  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+'''
+'''
+ * @file connectionManager.py
+ * @author Robert Taylor
+ * @date 2022-09-21
+'''
+
+import time
+import libcsp_py3 as libcsp
+from collections import defaultdict
+
+class ConnectionManager:
+    def __init__(self):
+        self.server_connection = defaultdict(dict)
+        self.rdp_timeout = 10000
```

Comment:
Interesting. Are we still entertaining the idea of using RDP some day?

Reply:
No, I could remove that. I left it because the getconnection method is so confusing and I copied it directly from the old code